libobs/obs-source.h

```diff
@@ -552,6 +552,10 @@ struct obs_source_info {
 	enum gs_color_space (*video_get_color_space)(
 		void *data, size_t count,
 		const enum gs_color_space *preferred_spaces);
+
+	/** Gets custom icons for dark and light themes */
+	const char *(*get_dark_icon)(void *type_data);
+	const char *(*get_light_icon)(void *type_data);
```

Comment:
Could these just be `const char *` set directly like `id`? Or is there is specific reason for this being a function?

Reply:
No, since we need to call the obs_module_file() function to get the path of the icons. Similar how get_name needs to call the obs_module_text() function.